accepted/00000-salt-ssh-with-salt-bundle.md

```diff
@@ -0,0 +1,77 @@
+So we don't need to maintain old salt codebases, but focus only on the latest one.
+
+# Detailed design
+[design]: #detailed-design
+
+## Uyuni roster
+
+We can use Uyuni roster module to provide roster data instead of creating flat roster files with Java. PR moving the logic to Uyuni roster on Java side: https://github.com/uyuni-project/uyuni/pull/4457
+
+This change can help to avoid creating temporary roster files (https://docs.saltproject.io/en/latest/topics/ssh/roster.html) and implements caching additionally.
+
+## Pre flight script
+
+Salt can execute a shell script on the salt-ssh client before the "real salt-ssh execution" starts.
+This script downloads the salt-bundle package from the appropriate bootstrap repository and extracts the virtual environment.
+By-default there is no way to pass additional parameters to pre fligt script running on the managed system.
+
+The "Pre flight script" calculates the bootstrap repository URL in a similar manner to the existing boostrap.sh script, based on the operating system and OS architecture of the system it is executed on.
+It requires one parameter from the Uyuni server: (host, port) (entry point) for SSH connections through tunnels.
+The salt codebase need to be modified to pass additional parameters, otherwise we have to generate pre flight script individually for each managed system.
+
+## Deploying Salt Bundle to the managed system
+
+We don't need to install Salt Bundle package on the managed system.
+We need to extract the virtual environment of the bundle from the package.
+Pre flight script is deploying it as described above.
+
+## Using proper salt codebase
+
+On deploying Salt Bundle with pre flight script, without modification salt-thin (`salt-thin` is a tarball with salt codebase created by the server to deploy salt on salt-ssh managed system: https://docs.saltproject.io/en/latest/ref/runners/all/salt.runners.thin.html) deployment, we will get two different codebases of salt deployed on the managed system (one codebase is from salt bundle package for the OS+arch and the other one is from salt-master delivered with salt-thin tarball).
+As the sources of these two codebases are different (update channel of the Server and client tools relevant for the system) these codebases are not in sync.
+We can prevent deploying salt-thin in case if Salt Bundle has been deployed already and use it as preferred salt codebase to handle salt-ssh session with on managed system side.
+In most cases the client tools channels are updating more frequently than the server and for the minions (not salt-ssh systems) we are using the salt codebase from client tools, it's better to use Salt Bundle codebase for salt-ssh systems.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+  * Salt Bundle consumes more space than salt-thin on managed system
```

Comment:
Is it possible to have a quantification, even approximate? Eg. a 20% increase vs. a 200% increase.

Reply:
current extracted `salt-thin` with 3 salt codebases (`3002.2` from the master, `py26-compat-salt` and `py27-compat-salt`) is about 90Mb (~30+ Mb per each salt codebase), extracted salt bundle slightly larger than 100 Mb. On removing py26- and py27- `salt-thin` will be ~40Mb.